bouncer/tests/delta_base_ingress.ts

```diff
@@ -0,0 +1,184 @@
+  numberOfNodes: 1 | 3 = 1,
+): Promise<void> {
+  const sourceAsset = 'Sol';
+  const amountFirstDeposit = '5';
+  const amountSecondDeposit = '1';
+
+  // Monitor swap events to make sure there is only one
+  let swapScheduledHandle = observeEvent('swapping:SwapScheduled', {
+    test: (event) => {
+      const data = event.data;
+
+      swapsWitnessed++;
+      console.log('Swap Scheduled found, swaps witnessed: ', swapsWitnessed);
+
+      if (swapsWitnessed > 1) {
+        throw new Error('More than one swaps were initiated');
+      }
+
+      const inputAmount = Number(data.inputAmount.replace(/,/g, ''));
+      if (
+        inputAmount > Number(amountToFineAmount(amountFirstDeposit, assetDecimals(sourceAsset)))
+      ) {
+        throw new Error(
+          'Swap input amount is greater than the first deposit ' + inputAmount.toString(),
+        );
+      }
+      return false;
+    },
+    abortable: true,
+  });
+
+  // Initiate swap from Solana
+  const swapParams = await testSwap(
+    sourceAsset,
+    'Eth',
+    undefined,
+    undefined,
+    undefined,
+    ' DeltaBasedIngress',
+    amountFirstDeposit.toString(),
+  );
+
+  await observeFetch(sourceAsset, swapParams.depositAddress);
+
+  await killEngines();
+  await startEngines(localnetInitPath, binariesPath, numberOfNodes);
+
+  // Wait to ensure no new swap is being triggered after restart.
+  console.log('Waiting for 30 seconds to ensure no swap is being triggered after restart');
+  await sleep(30000);
+  swapScheduledHandle.stop();
+
+  if (swapsWitnessed !== 1) {
+    throw new Error('No swap was initiated. Swaps witnessed: ' + swapsWitnessed);
+  }
+
+  // Kill the engine
+  await killEngines();
+
+  // Start another swap doing another deposit to the same address
+  const swapHandle = doPerformSwap(
+    swapParams,
+    `[${sourceAsset}->Eth DeltaBasedIngressSecondDeposit]`,
+    undefined,
+    undefined,
+    amountSecondDeposit,
+  );
+
+  swapScheduledHandle = observeEvent('swapping:SwapScheduled', {
+    test: (event) => {
+      const data = event.data;
+
+      swapsWitnessed++;
+      console.log('Swap Scheduled found, swaps witnessed: ', swapsWitnessed);
+
+      if (swapsWitnessed > 2) {
+        throw new Error('More than two swaps were initiated');
+      }
+      const inputAmount = Number(data.inputAmount.replace(/,/g, ''));
+      if (
+        inputAmount > Number(amountToFineAmount(amountSecondDeposit, assetDecimals(sourceAsset)))
+      ) {
+        throw new Error(
+          'Swap input amount is greater than the second deposit ' + inputAmount.toString(),
+        );
+      }
+      return false;
+    },
+    abortable: true,
+  });
+  await startEngines(localnetInitPath, binariesPath, numberOfNodes);
+
+  // Wait to ensure no additional new swap is being triggered after restart
+  // and check that swap completes.
+  await sleep(30000);
+  await swapHandle;
+  swapScheduledHandle.stop();
+
+  if (swapsWitnessed < 2) {
```

Comment:
Having this condition twice, here and inside SwapScheduled is redundant (and confusing imo).

Reply:
it's not the same check. Inside the `SwapScheduled` we check that there are no more than two saps and here we check that there not less than two. We are making sure that one swap was seen (aka final number of swaps == 2)